### PR TITLE
Fix overwriting os/arch when publishing to helix

### PR DESCRIPTION
**Pull request by jashook**
ptal @MattGal @RussKeldorph 

@wtgodbe @MattGal any suggestions on how to test this?

### CODE REVIEW

**Comment by RussKeldorph** [comment]
@chcosta @jcagme Can you help with "how to test" question above?

File: buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json [hunk 1]
```diff
@@ -122,7 +122,7 @@
...
+        "arguments": "helixpublish.proj /p:CloudDropAccessToken=$(CloudDropAccessToken) /p:CloudDropAccountName=$(CloudDropAccountName) /p:ContainerName=$(PB_ContainerName) /p:Platform=$(Architecture) /p:BuildType=$(PB_BuildType) /p:Rid=$(Rid) /p:CloudResultsAccountName=$(CloudResultsAccountName) /p:CloudResultsAccessToken=$(CloudResultsAccessToken) /p:TargetsWindows=$(TargetsWindows) /p:OverwriteOnUpload=true /p:Rid=$(Rid) /p:TargetQueues=\"$(TargetQueues)\" /p:TestProduct=$(TestProduct) /p:Branch=$(SourceBranch) /p:HelixApiAccessKey=$(HelixApiAccessKey) /p:HelixApiEndpoint=$(HelixApiEndpoint) /p:FilterToOSGroup=$(FilterToOSGroup) /p:FilterToTestTFM=$(FilterToTestTFM) /p:TimeoutInSeconds=1800 /p:HelixJobType=$(HelixJobType) /fileloggerparameters:Verbosity=diag;LogFile=helix.log",
```
Comment (MattGal) [inline buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json]: pretty sure the only thing /p:Rid=$(Rid) does, given that you have https://github.com/dotnet/coreclr/blob/d07f13c57203348d9ddb5a4bd001e00b0e30a242/buildpipeline/pipelines.json#L381, is prevent properties from setting it differently later in the build, since build properties and env vars get picked up automatically.  Doesn't break anything in this specific scenario AFAIk though, so it's fine.

**Comment by chcosta** [comment]
I'll examine the current state of coreclr builds and get back to you with testing instructions later this afternoon.

**Comment by MattGal** [comment]
@chcosta for my own understanding, why can't they just kick off a master build once this merges?

**Comment by chcosta** [comment]
That's certainly one way to test.  Another option, however, is to use the "PrivateRun" option of PipeBuild to validate the changes prior to merging into master.

**Comment by jcagme** [comment]
I like the "PrivateRun" option better

**Comment by chcosta** [comment]
@jashook, you can push your changes to a VSTS branch of the CoreClr Git repo, then queue a new build at https://devdiv.visualstudio.com/DevDiv/_build/index?definitionId=1832&_a=completed.  At queue time set the `PB_PrivateRun` property to `True` and `Branch` property to your branch name.  If  you need more explicit instructions, feel free to Skype me or stop by my office.

File: buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json (hunk 1 above)
Comment (wtgodbe) [reply buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json]: Why do you have `/p:Rid=$(Rid)` in here twice now?

File: tests/helixpublish.proj [hunk 2]
```diff
@@ -76,15 +76,15 @@
...
+        <RelativeBlobPath>$(Platform)$(BuildType)/%(Identity)</RelativeBlobPath>
       </HelixCorrelationPayloadFile>
       <HelixWorkItem>
         <Command Condition="'$(TargetsWindows)' == 'true'">$(HelixPythonPath) $(RunnerScript) --script %HELIX_WORKITEM_PAYLOAD%\runtests.cmd</Command>
         <Command Condition="'$(TargetsWindows)' != 'true'">chmod +x $HELIX_WORKITEM_PAYLOAD/runtests.sh &amp;&amp; $(HelixPythonPath) $(RunnerScript) --script $HELIX_WORKITEM_PAYLOAD/runtests.sh</Command>
         <PayloadFile>%(Identity)</PayloadFile>
         <WorkItemId>%(Filename)</WorkItemId>
         <TimeoutInSeconds>$(TimeoutInSeconds)</TimeoutInSeconds>
-        <RelativeBlobPath>$(Platform)-$(ConfigurationGroup)/Tests/$([System.String]::Copy('%(RecursiveDir)').Replace('\', '/'))%(Filename)%(Extension)</RelativeBlobPath>
+        <RelativeBlobPath>$(Platform)-$(BuildType)-$(Rid)/Tests/$([System.String]::Copy('%(RecursiveDir)').Replace('\', '/'))%(Filename)%(Extension)</RelativeBlobPath>
```
Comment (wtgodbe) [inline tests/helixpublish.proj]: What's the motivation for this change?

File: buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json (hunk 1 above)
Comment (MattGal) [reply buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json]: Good point Will :) Pretty sure it isn't even needed for 1 given it's a build parameter as well.

File: tests/helixpublish.proj (hunk 2 above)
Comment (MattGal) [reply tests/helixpublish.proj]: ConfigurationGroup is unset now, leading to uploading all tests for all platforms to paths like x64- and arm- folders.  This was causing massive file stomping, leading to almost all tests failing (it was a random race condition thing)

Comment (jashook) [reply tests/helixpublish.proj]: It is a little more confusing than this. As just having the configuration after the arch is also not sufficient enough to guarantee independent resources.

Comment (MattGal) [reply tests/helixpublish.proj]: @jashook was the variable not also unset?

Comment (jashook) [reply tests/helixpublish.proj]: Looks as if the variable has been under for a while. The cause instead seems to be linked to a change in pb_container name, which used to have the os.

**Comment by jashook** [comment]
Thanks Will. The private run failed after publishing failed. In order to save time I will remove rid, update and or and merge.
